extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -298,22 +270,23 @@ public void triggerEvent(String channelId, String event) {
     }
 
     /**
-     * Returns the scheduler for the Astro jobs
-     */
-    public ExpressionThreadPoolExecutor getScheduler() {
-        return scheduledExecutor;
-    }
-
-    /**
      * Adds the provided {@link Job} to the queue (cannot be {@code null})
      *
      * @return {@code true} if the {@code job} is added to the queue, otherwise {@code false}
      */
-    public boolean addJobToQueue(Job job) {
-        if (job != null) {
-            return scheduledJobs.add(job);
+    public void schedule(Job job, Calendar eventAt) {
+        monitor.lock();
+        try {
+            long sleepTime = eventAt.getTimeInMillis() - new Date().getTime();
```

Comment:
> we maybe can handle this easily as the astro calculations internally use julian dates,

It has been more about the drift of RTCs in embedded systems.
E.g. you realize that the system time is some minutes in the future or past and you change the system time to fix that situation (if you use OS features like adjtime or fix it by a jump is out of scope).

Okay, let's postpone that cases...